Userland/Services/SystemServer/Service.cpp

```diff
@@ -198,10 +198,11 @@ ErrorOr<void> Service::spawn(int socket_fd)
             TRY(Core::System::setenv("SOCKET_TAKEOVER"sv, socket_takeover_builder.string_view(), true));
         }
 
-        if (m_account.has_value() && m_account.value().uid() != getuid()) {
+        auto current_uid = getuid();
+        if (m_account.has_value() && m_account.value().uid() != current_uid) {
             auto& account = m_account.value();
-            if (auto error_or_void = account.login(); error_or_void.is_error()) {
-                dbgln("Failed to drop privileges (GID={}, UID={}), due to {}\n", account.gid(), account.uid(), error_or_void.error());
+            if (auto error_or_void = account.login(); error_or_void.is_error() && current_uid == 0) {
+                dbgln("Failed to drop privileges (tried to change to GID={}, UID={}), due to {}\n", account.gid(), account.uid(), error_or_void.error());
```

Comment:
It seems very security sensitive to just continue if _anything_ goes wrong with dropping privileges...

Reply:
Dropping privileges makes sense only when `SystemServer --user` is running as root. How otherwise we could change UID/GID when spawning a new service?